docs/reference/yaml-format-reference.md

```diff
@@ -123,6 +121,50 @@ recipeList:
 
 In this example, if a file matches `**/my.json` OR `**/your.json*` OR `**/our.json`, then the precondition has passed and the `ChangeKey` recipe will be applied to it.
 
+#### Precondition scope
+
+A common mistake new users make is to assume that if a precondition matches any file, then the recipe will apply to the entire repository.
+
+For example, attempting to use `FindPlugins` as a precondition for `CommonStaticAnalysis` to apply the static analysis fixes to only Gradle projects applying the sonar plugin:
+```yaml
+type: specs.openrewrite.org/v1beta/recipe
+name: org.sample.FlawedSonarStaticAnalysis
+displayName: Fix sonar issues 
+description: >- 
+  This recipe is intended to apply common static analysis issues only to gradle projects that apply the sonar plugin. 
+  In practice this precondition will limit changes to apply ONLY to build.gradle(.kts) files. 
+  But common static analysis issues are found within the sources, not the build scripts, so no changes will ever be made.
+preconditions:
+  - org.openrewrite.gradle.search.FindPlugins:
+      pluginId: org.sonarqube
```

Comment:
Perhaps @mike-solomon will have some ideas around this ; I'm usually hesitant to add "don't do this" into docs, as then folks happily copy paste the thing they shouldn't do and complain. I'd lead with or only show the thing they should do, and explain that not all recipes are a good fit separately in a warning block.

Reply:
I agree with your sentiment. Will rewrite this section a bit to hopefully get the same meaning across.